multithreading/assert.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
```

Comment:
I would remove this file (& the cpp) and just use regular `assert` (from `#include <cassert>`) instead of `aa_assert`